components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/codegen/ThrottlePolicyGenerator.java

```diff
@@ -42,24 +47,66 @@
  */
 public class ThrottlePolicyGenerator {
 
-    /**
-     * Generate ballerina and stream source for a given app and subs policies
-     *
-     * @param outPath              Destination file path to save generated source files. If not provided
-     *                             {@code definitionPath} will be used as the default destination path
-     * @param applicationPolicies  list of app policies
-     * @param subscriptionPolicies list of subs policies
-     * @throws IOException                  when file operations fail
-     * @throws BallerinaServiceGenException when code generator fails
-     */
-    public void generate(String outPath, List<ApplicationThrottlePolicyDTO> applicationPolicies,
-            List<SubscriptionThrottlePolicyDTO> subscriptionPolicies) throws IOException, BallerinaServiceGenException {
-        List<GenSrcFile> genFiles = generateApplicationPolicies(applicationPolicies);
+//    /**
+//     * Generate ballerina and stream source for a given app and subs policies
+//     *
+//     * @param outPath              Destination file path to save generated source files. If not provided
+//     *                             {@code definitionPath} will be used as the default destination path
+//     * @param applicationPolicies  list of app policies
+//     * @param subscriptionPolicies list of subs policies
+//     * @throws IOException                  when file operations fail
+//     * @throws BallerinaServiceGenException when code generator fails
+//     */
+//    public void generate(String outPath, List<ApplicationThrottlePolicyDTO> applicationPolicies,
+//            List<SubscriptionThrottlePolicyDTO> subscriptionPolicies) throws IOException, BallerinaServiceGenException {
+//        List<GenSrcFile> genFiles = new ArrayList<>();
+//        List<GenSrcFile> genAppFiles = generateApplicationPolicies(applicationPolicies);
+//
+//        if(genAppFiles != null){
+//            genFiles.addAll(genAppFiles);
+//        }
+//
+//        List<GenSrcFile> genSubsFiles = generateSubscriptionPolicies(subscriptionPolicies);
+//
+//        if(genSubsFiles != null){
+//            genFiles.addAll(genSubsFiles);
+//        }
+//
+//        GenSrcFile initGenFile = generateInitBal(applicationPolicies, subscriptionPolicies);
+//        genFiles.add(initGenFile);
+//        CodegenUtils.writeGeneratedSources(genFiles, Paths.get(outPath), true);
+//    }
 
-        List<GenSrcFile> genSubsFiles = generateSubscriptionPolicies(subscriptionPolicies);
-        genFiles.addAll(genSubsFiles);
+    //todo: remove unnecessary code segments
+    public void generate(String outPath, String projectName) throws IOException {
+
+        ApplicationThrottlePolicyListDTO applicationPolicies = restoreApplicationThrottlePolicy(projectName);
+        SubscriptionThrottlePolicyListDTO subscriptionPolicies = restoreSubscriptionThrottlePolicy(projectName);
+
+        if(applicationPolicies == null && subscriptionPolicies == null){
```

Comment:
Formatting issue. Add relevant spaces. Fix this issue in other files as well.

Reply:
Already fixed by a later added PR.